### DISCUSSION TITLE
ESP32 Ananlog Pins

### DISCUSSION THREAD

**Discussion by daniel3p4u** [arendst/Tasmota#19767] [Q&A] [answered]
Hi, i'm using tasmota (precompiled binary v13.1.0.4) with a ESP-WROOM-32 development board. I'm wondering why there are only four ADC GPIO pins (gpio34, gpio35, gpio36, gpio39)  available in Tasmota. Following the ESP-32 datasheet there are a lot more analog pins. I've tested them with a small test script. In tasmota they are selectable and show up in web interface but always show the same value. Is there any way to bring these pins into tasmota?

**Comment by sfromis** [comment, marked as answer]
One reason is that the ADC2 block is not workable with Wifi, and therefore not supported by Tasmota. But you *can* use the pins of the ADC1 block, assuming that you have a board where gpio 32-39 are broken out. That's not the case on all dev boards.
